### PR TITLE
update to latest glutin

### PR DESCRIPTION
**Pull request by happydpc**

### CODE REVIEW

File: src/render/mod.rs [hunk 1]
```diff
@@ -538,13 +539,14 @@ impl Renderer {
...
+        context: glutin::ContextBuilder<NotCurrent>,
         event_loop: &glutin::EventsLoop,
         source: &source::Set,
-    ) -> (Self, glutin::GlWindow, Factory) {
+    ) -> (Self, glutin::WindowedContext<PossiblyCurrent>, Factory) {
         use gfx::texture as t;
 
-        let (window, device, mut gl_factory, out_color, out_depth) = gfx_window_glutin::init(builder, context, event_loop).unwrap();
+        let (_windowedContext, device, mut gl_factory, out_color, out_depth) = gfx_window_glutin::init(builder, context, event_loop).unwrap();
+        let window = _windowedContext.window();
```
Comment (vitvakatu) [inline src/render/mod.rs]: Please don't use underscore in front of regular variables. `windowedContext` doesn't need it

File: src/window.rs [hunk 2]
```diff
@@ -111,6 +112,8 @@ impl Builder {
             .with_title(self.title.clone());
 
         let context = glutin::ContextBuilder::new()
+            .with_gl_profile(GlProfile::Core)
+            .with_gl(GlRequest::Latest)
```
Comment (vitvakatu) [inline src/window.rs]: I'd rather prefer to see a specific version here. @kvark what do you think?

File: src/window.rs [hunk 3]
```diff
@@ -147,12 +150,13 @@ impl Builder {
             try_override!(basic, gouraud, pbr, phong, quad, shadow, skybox, sprite,);
         }
 
-        let (renderer, window, mut factory) = Renderer::new(builder, context, &event_loop, &source_set);
-        let dpi = window.get_hidpi_factor();
+        let (renderer, windowedContext, mut factory) = Renderer::new(builder, context, &event_loop, &source_set);
+        let dpi = windowedContext.window().get_hidpi_factor();
         let scene = factory.scene();
+//        let window = windowedContext.window();
```
Comment (vitvakatu) [inline src/window.rs]: Please remove commented code

File: src/window.rs [hunk 4]
```diff
@@ -191,16 +195,17 @@ impl Window {
             input.reset();
         }
 
-        self.window.swap_buffers().unwrap();
-        let window = &self.window;
+//        self.windowedContext.window().swap_buffers().unwrap();
```
Comment (vitvakatu) [inline src/window.rs]: Here as well

File: src/window.rs [hunk 5]
```diff
@@ -242,17 +247,18 @@ impl Window {
...
+        let size = self.windowedContext
+            .window()
             .get_inner_size()
             .expect("Can't get window size")
             .to_physical(self.dpi);
         [size.width as f32, size.height as f32].into()
     }
 
-    /// Returns underlaying `glutin::GlWindow`.
+    /// Returns underlaying `glutin::Window`.
```
Comment (vitvakatu) [inline src/window.rs]:
```suggestion
    /// Returns underlaying `glutin::WindowedContext`.
```

File: src/window.rs (hunk 2 above)
Comment (kvark) [reply src/window.rs]: Some older GPUs only support GL 2.1, while some macs do *not* support older GL 2x. So if we are completely pedantic, we can't have a specific version requirement (even though, practically, GL 3.2 would fit most cases). So having `Latest` here is fine.

**Comment by happydpc** [comment]
Sorry for these small problems. Will three.rs plan to support gfx-hal, because I found that the opengl was not that enough. How about supporting rendy ?

**Comment by kvark** [comment]
@happydpc I think it would make more sense to base on wgpu-rs for this project instead of gfx/rendy. See https://github.com/three-rs/three/issues/83#issuecomment-529161223 and #222

**Comment by happydpc** [comment]
Yes, that's true. I had a try on wgpu-rs yesterday, it will be cool if running on web browsers in the future instead of by poor WebGL library.  I am planning to port some examples in [https://github.com/SaschaWillems/Vulkan](url) with wgpu-rs.

**Comment by kvark** [comment]
Bors r=vitvakatu

**Pull request merged**